HW_2.py

```diff
@@ -0,0 +1,438 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Задание 1 (2 балла)
+
+# Напишите класс `MyDict`, который будет полностью повторять поведение обычного словаря, за исключением того, что при итерации мы должны получать и ключи, и значения.
+# 
+# **Модули использовать нельзя**
+
+# In[32]:
+
+
+# Ваш код здесь
+class MyDict(dict):
+    def __iter__(self):
+        for key in super().keys():
+            yield (key, self[key])
+
+
+# In[33]:
+
+
+dct = MyDict({"a": 1, "b": 2, "c": 3, "d": 25})
+for key, value in dct:
+    print(key, value)   
+
+
+# In[34]:
+
+
+for key, value in dct.items():
+    print(key, value)
+
+
+# In[35]:
+
+
+for key in dct.keys():
+    print(key)
+
+
+# In[36]:
+
+
+dct["c"] + dct["d"]
+
+
+# # Задание 2 (2 балла)
+
+# Напишите функцию `iter_append`, которая "добавляет" новый элемент в конец итератора, возвращая итератор, который включает изначальные элементы и новый элемент. Итерироваться по итератору внутри функции нельзя, то есть вот такая штука не принимается
+# ```python
+# def iter_append(iterator, item):
+#     lst = list(iterator) + [item]
+#     return iter(lst)
+# ```
+# 
+# **Модули использовать нельзя**
+
+# In[50]:
+
+
+def iter_append(iterator, item):
+    # Ваш код здесь
+    for element in iterator:
```

Comment:
Можно было просто так
```suggestion
    yield from iterator
```